packages/block-library/src/query-total/index.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Server-side rendering of the `core/query-total` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `query-total` block on the server.
+ *
+ * @since 6.8.0
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string The rendered block content.
+ */
+function render_block_core_query_total( $attributes, $content, $block ) {
+	global $wp_query;
+	$wrapper_attributes = get_block_wrapper_attributes();
+	if ( isset( $block->context['query']['inherit'] ) && $block->context['query']['inherit'] ) {
+		$query_to_use = $wp_query;
+		$current_page = max( 1, get_query_var( 'paged', 1 ) );
+	} else {
+		$page_key     = isset( $block->context['queryId'] ) ? 'query-' . $block->context['queryId'] . '-page' : 'query-page';
+		$current_page = isset( $_GET[ $page_key ] ) ? (int) $_GET[ $page_key ] : 1;
+		$query_to_use = new WP_Query( build_query_vars_from_query_block( $block, $current_page ) );
+	}
+
+	$max_rows       = $query_to_use->found_posts;
+	$posts_per_page = $query_to_use->get( 'posts_per_page' );
+
+	// Calculate the range of posts being displayed.
+	$start = ( $current_page - 1 ) * $posts_per_page + 1;
+	$end   = min( $start + $posts_per_page - 1, $max_rows );
+
+	// Prepare the display based on the `displayType` attribute.
+	$output = '';
+	switch ( $attributes['displayType'] ) {
+		case 'total-results':
+		default:
```

Comment:
The case for `range-display` needs to be moved before the `default` clause as it will never be reachable otherwise 😄

Reply:
I missed that in all the excitement! 😆